prefix-path with minimal changes (#6897)
plone/volto: docs/source/deploying/prefix-path.md

---
myst:
  html_meta:
    "description": "Serve Volto on a subpath"
    "property=og:description": "Serve Volto on a subpath"
    "property=og:title": "Serve Volto on a subpath"
    "keywords": "Volto, Plone, path, prefix, virtual hosting"
---

# Serve Volto on a subpath

> Volto can be served on a subpath, e.g. `https://mysite.com/subpath`.

Reviewer: From chatgpt:

> What would you call a feature where a website is served at a location other than the root path?

Good question — there are a few terms that could describe that situation depending on context. Here are the most common ones:

🧭 General web terminology

Subpath deployment – a general and modern way to say the site is hosted under a path, e.g. example.com/myapp/ instead of example.com/.

“The app supports subpath deployment.”

Subdirectory deployment – similar meaning; emphasizes the directory structure of the URL.

“The docs site runs in a subdirectory.”

Non-root path or non-root mount – explicitly says it’s not at /.

“The site is mounted on a non-root path.”

⚙️ In framework or server contexts

Mount point – used when describing how an app is attached to a web server or WSGI pipeline.

“The Plone instance is mounted at /intranet.”
(Common in Zope, Flask, Express, etc.)

Base path – sometimes refers to the configured URL prefix the app assumes.

“The frontend’s base path is /myapp.”

Context path – Java EE / servlet term, but sometimes borrowed elsewhere.

“The application’s context path is /portal.”

If you’re writing documentation for developers, “subpath deployment” or “mounted under a subpath” are the clearest and most framework-neutral phrases.

> When this is enabled, both content and static assets are served on the subpath, which means a different website can be served at the root path.

Reviewer 2: I think this is more explicit.

Can we serve multiple sites on multiple subpaths, or is it limited to just one?

```suggestion
When this is enabled, both content and static assets are served on the subpath.
This means different websites can be served at both the root path and the subpath.
```

Reviewer: @stevepiercy I got caught on the wording of your second sentence (if they're different, then it's odd to use the word "both"). We're also making it sound like maybe both of these can be served through one Volto instance, but no, I'm saying it leaves the root path free in case you want to route it to different software via a frontend webserver.

How about:
```suggestion
When this is enabled, both content and static assets are served on the subpath.
This leaves the root path free in case you want to route it to a different website.
```

> To use a subpath, run Volto with the `RAZZLE_PREFIX_PATH` environment variable set:

Reviewer 2: ```suggestion
To use a subpath, run Volto with the `RAZZLE_PREFIX_PATH` environment variable set to the subpath.
```

```shell
RAZZLE_PREFIX_PATH=/subpath make start
```

> If you are building bundles to run in production mode, the environment variable must be set at both build and run time:

Reviewer 2: ```suggestion
If you build bundles to run in production mode, you must set the environment variable at both build and run times as shown.
```
